src/argus/filter/queryset_filters.py

```diff
@@ -0,0 +1,128 @@
+    if filter_open is False:
+        fitting_incidents = fitting_incidents.closed()
+    if filter_acked is True:
+        fitting_incidents = fitting_incidents.acked()
+    if filter_acked is False:
+        fitting_incidents = fitting_incidents.not_acked()
+    if filter_stateful is True:
+        fitting_incidents = fitting_incidents.stateful()
+    if filter_stateful is False:
+        fitting_incidents = fitting_incidents.stateless()
+    return fitting_incidents.distinct()
+
+
+def _incidents_fitting_maxlevel(incident_queryset, filterblob: FilterBlobType):
+    maxlevel = filterblob.get("maxlevel", None)
+    if not maxlevel:
+        return incident_queryset.distinct()
+    return incident_queryset.filter(level__lte=maxlevel).distinct()
+
+
+def _incidents_fitting_event_types(incident_queryset, filterblob: FilterBlobType):
+    event_types = filterblob.get("event_types", [])
+    if not event_types:
+        return incident_queryset.distinct()
+    return incident_queryset.prefetch_related("events").filter(events__type__in=event_types).distinct()
+
+
+class QuerySetFilter:
+    @staticmethod
+    def filtered_incidents(filterblob: FilterBlobType, incident_queryset=None):
+        if not incident_queryset:
+            incident_queryset = Incident.objects.all()
+        filterblob = filterblob.copy()
+        fw = FilterWrapper(filterblob)
+        if fw.is_empty:
+            return Incident.objects.none().distinct()
+        filtered_by_event_types = _incidents_fitting_event_types(incident_queryset, filterblob)
+        filtered_by_source = _incidents_with_source_systems(incident_queryset, filterblob)
+        filtered_by_tags = _incidents_with_tags(incident_queryset, filterblob)
+        filtered_by_tristates = _incidents_fitting_tristates(incident_queryset, filterblob)
+        filtered_by_maxlevel = _incidents_fitting_maxlevel(incident_queryset, filterblob)
+
+        return (
+            filtered_by_event_types
+            & filtered_by_source
+            & filtered_by_tags
+            & filtered_by_tristates
+            & filtered_by_maxlevel
+        )
+
+    @classmethod
+    def incidents_by_filter(cls, incident_queryset, filter: Filter):
+        "Returns all incidents that are included in the filter instance"
+        return cls.filtered_incidents(filter.filter, incident_queryset).all()
+
+    @classmethod
+    def incidents_by_filter_pk(cls, incident_queryset, filter_pk: int):
+        """
+        Returns all incidents that are included in the filter with the given primary
+        key
+
+        If no filter with that pk exists it returns no incidents
+        """
+        filtr = Filter.objects.filter(pk=filter_pk).first()
+
+        if not filtr:
+            return incident_queryset.none()
+
+        return cls.incidents_by_filter(incident_queryset, filtr)
+
+    @classmethod
+    def incidents_by_notificationprofile(cls, incident_queryset, notificationprofile):
+        if not incident_queryset:
```

Comment:
```suggestion
        if incident_queryset is None:
```
otherwise this will also be true if for some reason the given query set is empty